src/internal/m365/onedrive/restore.go

```diff
@@ -1107,3 +1182,70 @@ func AugmentRestorePaths(
 
 	return paths, nil
 }
+
+type PostDriveAndGetRootFolderer interface {
+	PostDriver
+	GetRootFolderer
+}
+
+// ensureDriveExists looks up the drive by its id.  If no drive is found with
+// that ID, a new drive is generated with the same name.  If the name collides
+// with an existing drive, a number is appended to the drive name.  Eg: foo ->
+// foo 1.  This will repeat as many times as is needed.
+// Returns the root folder of the drive
+func ensureDriveExists(
+	ctx context.Context,
+	pdagrf PostDriveAndGetRootFolderer,
+	caches *restoreCaches,
+	drivePath *path.DrivePath,
+	protectedResourceID, driveName string,
+) error {
+	driveID := drivePath.DriveID
+
+	// the drive might already be cached
+	if _, ok := caches.DriveIDToDriveInfo[driveID]; ok {
+		return nil
+	}
+
+	var (
+		newDriveName = driveName
+		newDrive     models.Driveable
+		err          error
+		i            = 1
+	)
+
+	if _, ok := caches.DriveNameToDriveInfo[newDriveName]; ok {
+		newDriveName = fmt.Sprintf("%s %d", driveName, 1)
```

Comment:
nit: this could take `i` as a param, instead of 1

Reply:
switched it around so that i is only in the for loop handler, instead of being a variable outside of it.  Makes it clearer how it's being used.